support/utils/functions_test.go

```diff
@@ -36,3 +36,90 @@ func TestDedupe(t *testing.T) {
 		})
 	}
 }
+
+func TestToMapStringInterface_SuccessMap(t *testing.T) {
+	success := map[string]interface{}{
+		"test": true,
+	}
+	m, e := ToMapStringInterface(success)
+	assert.Equal(t, success, m)
+	assert.NoError(t, e)
+}
+
+func TestToMapStringInterface_SuccessStruct(t *testing.T) {
+	type NonEmptyStruct struct {
+		Value int `json:"value"`
+	}
+
+	type SuccessStruct struct {
+		True           bool
+		False          bool
+		EmptyString    string
+		NonEmptyString string
+		Int            int
+		Float          float64
+		EmptyStruct    struct{}
+		NonEmptyStruct NonEmptyStruct
+		EmptyInts      []int
+		NonEmptyInts   []int
+	}
+
+	success := SuccessStruct{
+		True:           true,
+		False:          false,
+		EmptyString:    "",
+		NonEmptyString: "hello world",
+		Int:            1,
+		Float:          1.0,
+		EmptyStruct:    struct{}{},
+		NonEmptyStruct: NonEmptyStruct{1},
+		EmptyInts:      []int{},
+		NonEmptyInts:   []int{1, 2},
+	}
+
+	want := map[string]interface{}{
+		"EmptyInts":      []interface{}{},
+		"EmptyString":    "",
+		"EmptyStruct":    map[string]interface{}{},
+		"False":          false,
+		"Float":          1.,
+		"Int":            1.,
```

Comment:
@debnil why is the `Int` field showing up as a `float`?
is this an error somewhere and we cannot convert ints using this approach?

Reply:
This isn't an error, it's how JSON works. The standard only has numbers, and the `json` package assumes float64 when you haven't defined anything else (just Unmarshaling to an `interface{}` rather than a custom `struct`) - which is what we are doing. So, this is expected behavior. I've added a comment documenting this.

I think this is fine for our purposes. It would be bad if we were going from float to int (i.e., 64-bit to 32-bit precision). But, since we won't lose precision through this approach, I'm fine with it. It's good that we have a test that documents this behavior.

Reply:
yes, that's fair and best to document this in a comment. works for me 👍